### PR TITLE
operator: Add optional leader-election on startup

### PR DESCRIPTION
**Pull request by bison**
Adds leader-election on startup using a ConfigMap.

### CODE REVIEW

File: pkg/operator/config.go [hunk 1]
```diff
@@ -67,6 +97,24 @@ func ConfigFromEnvironment() *Config {
 		config.WatchNamespace = watchNamespace
 	}
 
+	if leaderElection, ok := os.LookupEnv("LEADER_ELECTION"); ok {
+		le, err := strconv.ParseBool(leaderElection)
+		if err != nil {
+			le = DefaultLeaderElection
+			glog.Errorf("Error parsing LEADER_ELECTION environment variable: %v", err)
```
Comment (frobware) [inline pkg/operator/config.go]: Is the intention here that this falls through and `le` is (possibly) `false` (depending on implementation)?

Comment (bison) [reply pkg/operator/config.go]: If the environment variable is present, but fails to parse, we log the error and use the default value. That happens to be `true` at the moment.

Comment (frobware) [reply pkg/operator/config.go]: Ah, sorry. Missed the `le = DefaultLeaderElection`. Somehow!

**Pull request merged**